prime-router/src/main/kotlin/cli/tests/BasicTests.kt

```diff
@@ -66,45 +72,55 @@ class Ping : CoolTest() {
 }
 
 /**
- * End to end test for the universal pipeline. This test should read in a submission, verify it reached all steps
- * (receive, convert, route, translate, batch, send) and that it generated the correct lineage records.
+ * An End-to-End test for the Universal Pipeline. This test posts a report and verifies that the file was sent to
+ * the expected receiver(s), then verifies the contents of the file match the expected contents.
+ * This test runs through four scenarios:
+ * - A report that is submitted as HL7 and is sent as HL7 (Topic: FULL-ELR).
+ * - A report that is submitted as HL7 and is sent as FHIR (Topic: FULL-ELR).
+ * - A report that is submitted as HL7 and is sent as HL7 but has the send original flag (Topic: ELIMS).
+ * - A report that is submitted as FHIR and is sent as FHIR (Topic: FULL-ELR).
  */
 class End2EndUniversalPipeline : CoolTest() {
```

Comment:
Right now the test is set up in such a way where it will fail early if any one test failed, but it would be better if we could run all of them. This would lead to faster turn around if more than one test was failing?

Reply:
Good idea. Currently the only spot that would cause the whole test to not run is if one of the reports never got a `overallStatus` of `Delivered`. I've changed that to log as a failure and continue the test. https://github.com/CDCgov/prime-reportstream/pull/13679/commits/caafe52d3289e87769aaae7bb31f24f14f6d6c55